File: backend/app/schemas/prompt.py
"""Prompt configuration schemas."""

from datetime import datetime

from pydantic import BaseModel, Field


class PromptConfig(BaseModel):
    """Prompt configuration model."""

    key: str = Field(..., description='Prompt identifier key')
    content: str = Field(..., description='Prompt content')
    description: str | None = Field(None, description='Prompt description')


class PromptConfigUpdate(BaseModel):
    """Prompt configuration update model."""

    content: str = Field(..., min_length=1, description='New prompt content')


class KnowledgeProfileResponse(BaseModel):
    """Knowledge profile response model."""

    status: str = Field(..., description='Profile status: building, ready, failed')
    major_topics: list[str] = Field(default_factory=list, description='Major knowledge topics')
    high_frequency_entities: list[str] = Field(default_factory=list, description='High frequency entities')
    high_frequency_relations: list[str] = Field(default_factory=list, description='High frequency relations')
    recent_focuses: list[str] = Field(default_factory=list, description='Recent knowledge focuses')
    rendered_overlay: str = Field('', description='Rendered prompt overlay text')
    updated_at: datetime | None = Field(None, description='Last update time')
    error_message: str | None = Field(None, description='Error message if failed')


class ComposedPromptResponse(BaseModel):
    """Composed system prompt response model."""

    base_prompt: str = Field(..., description='Base system prompt')
    overlay: str = Field('', description='Dynamic knowledge profile overlay')
    composed_prompt: str = Field(..., description='Final composed system prompt')
    profile_status: str | None = Field(None, description='Knowledge profile status')

File: frontend/src/types/prompt.ts
export interface PromptConfig {
  key: string
  content: string
  description: string | null
}

export interface PromptConfigUpdate {
  content: string
}

export interface KnowledgeProfile {
  status: string
  major_topics: string[]
  high_frequency_entities: string[]
  high_frequency_relations: string[]
  recent_focuses: string[]
  rendered_overlay: string
  updated_at: string | null
  error_message: string | null
}

export interface ComposedPrompt {
  base_prompt: string
  overlay: string
  composed_prompt: string
  profile_status: string | null
}
